solr/src/main/scala/akka/stream/alpakka/solr/SolrFlowStage.scala

```diff
@@ -8,51 +8,164 @@ import java.net.{ConnectException, SocketException}
 
 import akka.NotUsed
 import akka.stream.alpakka.solr.SolrFlowStage.{Finished, Idle, Sending}
-import akka.stream.stage.{GraphStage, InHandler, OutHandler, TimerGraphStageLogic}
-import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.stream.stage._
+import akka.stream._
 import org.apache.http.NoHttpResponseException
 import org.apache.solr.client.solrj.SolrClient
+import org.apache.solr.client.solrj.response.UpdateResponse
 import org.apache.solr.common.{SolrException, SolrInputDocument}
 
-import scala.collection.JavaConverters._
-import scala.collection.mutable
-import scala.concurrent.Future
+import scala.annotation.tailrec
 import scala.util.control.NonFatal
+import scala.collection.JavaConverters._
 
-object IncomingMessage {
-  // Apply method to use when not using passThrough
+private object IncomingMessage {
```

Comment:
Why did this become `private`?

I see you're creating different operations depending on the parameter lists, that is very confusing.
Please, have a look at my proposed changes to the Elasticsearch connector, as well: https://github.com/akka/alpakka/pull/1179 (I'm not saying you should do the whole Alpakka 1.0 thing in this PR.)